exchanges/gateio/gateio_types.go

```diff
@@ -598,13 +598,19 @@ type Trade struct {
 
 // Candlestick represents candlestick data point detail.
 type Candlestick struct {
-	Timestamp      time.Time
-	QuoteCcyVolume float64
-	ClosePrice     float64
-	HighestPrice   float64
-	LowestPrice    float64
-	OpenPrice      float64
-	BaseCcyAmount  float64
+	Timestamp      types.Time
+	QuoteCcyVolume types.Number
+	ClosePrice     types.Number
+	HighestPrice   types.Number
+	LowestPrice    types.Number
+	OpenPrice      types.Number
+	BaseCcyAmount  types.Number
+	WindowClosed   string
+}
+
+// UnmarshalJSON parses kline data from a JSON array into Candlestick fields.
+func (c *Candlestick) UnmarshalJSON(data []byte) error {
+	return json.Unmarshal(data, &[8]any{&(c.Timestamp), &(c.QuoteCcyVolume), &(c.ClosePrice), &(c.HighestPrice), &(c.LowestPrice), &(c.OpenPrice), &(c.BaseCcyAmount), &(c.WindowClosed)})
```

Comment:
I will stop mentioning this; you can remove all the brackets across this PR as they are not needed.